src/modules/account/contexts/ActionFieldContext.tsx

```diff
@@ -0,0 +1,35 @@
+import { FC, createContext, useContext, useState } from 'react';
+
+type ActionFieldContextValue = {
+  isShowingMainContent: boolean;
+  showMainContent: () => void;
+  showAlternativeContent: () => void;
+  toggleContent: () => void;
+};
+
+const ActionFieldContext = createContext<ActionFieldContextValue | null>(null);
+
+export const useActionField = () => {
+  const actionFieldContextValue = useContext<ActionFieldContextValue | null>(
+    ActionFieldContext
+  );
+  if (!actionFieldContextValue) throw Error('No matching AuthProvider');
+  return actionFieldContextValue;
+};
+
+export const ActionFieldProvider: FC = ({ children }) => {
+  const [isShowingMainContent, setIsShowingMainContent] = useState(false);
+
+  return (
+    <ActionFieldContext.Provider
+      value={{
```

Comment:
`useMemo` here. The only time `useMemo` is not necessary is when it is at the root of the app, which means it does not likely trigger a re-render because of another component.